File: c1/p7/ans.py
# print((lambda n, m, litter, food, dist_list: (type([dist_list.append([((litter[i][0] - food[j][0]) ** 2 + (litter[i][1] - food[j][1]) ** 2), litter[i], food[j]]) if (dist_list[len(dist_list) - 1][0] if len(dist_list) > 0 else -1) < ((litter[i][0] - food[j][0]) ** 2 + (litter[i][1] - food[j][1]) ** 2) else None for i in range(n) for j in range(m)]) == type([])) * (lambda dist_list: f"{dist_list[-1][1][0]} {dist_list[-1][1][1]}\n{dist_list[-1][2][0]} {dist_list[-1][2][1]}")(dist_list)) (*(lambda n, m, input: ([n, m, [list(map(int, input().split())) for _ in range(n)], [list(map(int, input().split())) for _ in range(m)], []]))(*map(int, input().split()), __import__("sys").stdin.readline)))

n, m = map(int, input().split())
litter = [list(map(int, input().split())) for _ in range(n)]
food = [list(map(int, input().split())) for _ in range(m)]

def dist(x: tuple, y: tuple) -> float:
    return ((x[0] - y[0]) ** 2 + (x[1] - y[1]) ** 2) ** 0.5
    
ans_dist = -1
lans, fans = (0, 0), (0, 0)
for i in range(n):
    for j in range(m):
        if dist(litter[i], food[j]) > ans_dist:
            ans_dist = dist(litter[i], food[j])
            lans, fans = litter[i], food[j]
print(*lans)
print(*fans)

File: c1/p7/ans.cpp
#include <bits/stdc++.h>
#define pii pair<int, int>
#define fi first
#define se second
using namespace std;

const int MAXSZ = 1e4 + 5;

int n, m;
double ans = -1;
pii litter[MAXSZ], food[MAXSZ];
pii lans, fans;

double dist(pii a, pii b) {
    return sqrt(((a.fi - b.fi) * (a.fi - b.fi)) + ((a.se - b.se) * (a.se - b.se)));
}

signed main() {

    ios_base::sync_with_stdio(0);cin.tie(0);

    cin >> n >> m;
    for (int i = 0; i < n; i ++) {
        cin >> litter[i].fi >> litter[i].se;
    }
    for (int i = 0; i < m; i ++) {
        cin >> food[i].fi >> food[i].se;
    }

    for (int i = 0; i < n; i ++) {
        for (int j = 0; j < m; j ++) {
            if (dist(litter[i], food[j]) > ans) {
                ans = dist(litter[i], food[j]);
                lans = litter[i];
                fans = food[j];
            }
        }
    }
    cout << lans.fi << " " << lans.se << "\n";
    cout << fans.fi << " " << fans.se << "\n";
    
    return 0;
}